modules/core/src/smithy4s/schema/Alt.scala

```diff
@@ -82,7 +91,11 @@ object Alt {
       ): G[U] = {
         val precompiledAlts =
           precompile.toPolyFunction
-            .unsafeCache(alts.map(smithy4s.Existential.wrap(_)))
+            .unsafeCacheBy[String](
+              alts.map(smithy4s.Existential.wrap(_)),
+              (alt: Existential[Alt[F, U, *]]) =>
```

Comment:
I've been away from polymorphic code so long that I almost forgot what `*` means here...